packages/tokens-studio-for-figma/src/app/asyncMessageHandlers/startup.tsx

```diff
@@ -1,27 +1,68 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { createRoot } from 'react-dom/client';
 import * as Sentry from '@sentry/react';
-import { Provider } from 'react-redux';
+import { Provider, useStore } from 'react-redux';
 import i18next from 'i18next';
 import * as Tooltip from '@radix-ui/react-tooltip';
-import { AsyncMessageChannelHandlers } from '@/AsyncMessageChannel';
-import { AsyncMessageTypes } from '@/types/AsyncMessages';
+
+import { AsyncMessageChannel } from '@/AsyncMessageChannel';
+import { AsyncMessageTypes, StartupMessage } from '@/types/AsyncMessages';
 import { ErrorFallback } from '../components/ErrorFallback';
-import { store } from '../store';
+import { RootState, store } from '../store';
 import { AppContainer } from '../components/AppContainer';
+import PreviewApp from './preview';
+import FigmaLoading from '../components/FigmaLoading';
+
+// eslint-disable-next-line
+const PREVIEW_ENV = process.env.PREVIEW_ENV;
+
+const StartupApp = () => {
+  const [params, setParams] = useState<StartupMessage | null>(null);
+  const rootStore = useStore<RootState>();
+  const state = rootStore.getState();
```

Comment:
Could this not be:
```suggestion
  const state = store.getState();
```

?